src/root.tsx

```diff
@@ -332,13 +332,15 @@ function isVNode(maybeNode: ComponentChild): maybeNode is VNode<unknown> {
   );
 }
 
-function childrenToTree(children: ComponentChild, root: Root<any>) {
-  return array(children).reduce(
-    (accumulator: {children: NodeTree, descendants: NodeTree}, next: ComponentChild) => {
-      accumulator.children.push(buildElementWrappers(next, root)[0]);
-      accumulator.descendants.push(...buildElementWrappers(next, root));
-      return accumulator;
-    },
-    {children: [] as NodeTree, descendants: [] as NodeTree},
-  );
+function childrenToTree(inputChildren: ComponentChild, root: Root<any>) {
+  const children: NodeTree = [];
+  const descendants: NodeTree = [];
+  
+  for (const child of array(inputChildren)) {
```

Comment:
In this case we actually care more about the machine than the human so we opt to put less functions on the stack where we can (saw a slight increase in the depth I could test with this change)